### PR TITLE
Implement readOnly property

### PR DESCRIPTION
**Pull request by hfiguiere**
-also fix event handling on disabled: we could still toggle.

### CODE REVIEW

**Comment by hfiguiere** [comment]
readOnly syntax is taken from rc-input-number.

**Comment by hfiguiere** [comment]
The idea is that readOnly is like disabled, but without non-disabled style, to display but not allow editing.

File: src/Switch.jsx [hunk 1]
```diff
@@ -20,20 +21,26 @@ const Switch = React.createClass({
...
     } else {
       checked = !!props.defaultChecked;
     }
+    if ('readOnly' in props) {
+      readOnly = !!props.readOnly;
+    }
     return {
-      checked,
+      checked: checked,
+      readOnly: readOnly,
```
Comment (afc163) [inline src/Switch.jsx]: this.state.readOnly is unnecessary, use this.props.readOnly directly.

**Comment by benjycui** [comment]
Conflict, and no response for a long time, close now.

Actually, I think you can just use `<Switch disabled />` or `<Switch checked />`

**Pull request closed without merging**